components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3566,12 +3560,7 @@ where
             // [entries...][the entry at `compact_idx`][the last entry][new compaction entry]
             //             |-------------------- entries will be left ----------------------|
             return;
-        } else if replicated_idx - first_idx < self.ctx.cfg.raft_log_gc_threshold
-            && self.fsm.skip_gc_raft_log_ticks < self.ctx.cfg.raft_log_reserve_max_ticks
```

Comment:
Looks like we can also cleanup the `raft_log_reserve_max_ticks` config?

Reply:
Yes. after this pr, `raft_log_reserve_max_ticks` won't be used. But it was listed in the `config-template`. We should keep it for compatibility, but hide it in the doc.

Reply:
It is okay to remove stale configs. TiKV will simply ignore non-existing configs.